applications/RANSApplication/python_scripts/formulations/monolithic_vms/monolithic_velocity_pressure_rans_formulation.py

```diff
@@ -0,0 +1,252 @@
+        Kratos.VariableUtils().AddDof(Kratos.PRESSURE, Kratos.REACTION_WATER_PRESSURE, base_model_part)
+
+        Kratos.Logger.PrintInfo(self.GetName(), "Added dofs.")
+
+    def PrepareModelPart(self):
+        self.monolithic_model_part = CreateRansFormulationModelPart(
+            self,
+            "VMS",
+            self.condition_name)
+        Kratos.Logger.PrintInfo(self.GetName(), "Created formulation model part.")
+
+    def Initialize(self):
+        model_part = self.GetBaseModelPart()
+        CalculateNormalsOnConditions(model_part)
+
+        process_info = model_part.ProcessInfo
+        bossak_alpha = process_info[Kratos.BOSSAK_ALPHA]
+
+        settings = self.GetParameters()
+
+        if (self.IsPeriodic()):
+            if (self.domain_size == 2):
+                periodic_variables_list = [Kratos.VELOCITY_X, Kratos.VELOCITY_Y, Kratos.PRESSURE]
+            else:
+                periodic_variables_list = [Kratos.VELOCITY_X, Kratos.VELOCITY_Y, Kratos.VELOCITY_Z, Kratos.PRESSURE]
+            InitializePeriodicConditions(model_part,
+                                         self.monolithic_model_part,
+                                         periodic_variables_list)
+
+        conv_criteria = GetKratosObjectType("MixedGenericCriteria")(
+            [(Kratos.VELOCITY, settings["relative_velocity_tolerance"].GetDouble(), settings["absolute_velocity_tolerance"].GetDouble()),
+             (Kratos.PRESSURE, settings["relative_pressure_tolerance"].GetDouble(), settings["absolute_pressure_tolerance"].GetDouble())])
+
+        if self.is_steady_simulation:
+            scheme = GetKratosObjectType("ResidualBasedSimpleSteadyScheme")(
+                settings["velocity_relaxation"].GetDouble(),
+                settings["pressure_relaxation"].GetDouble(),
+                self.domain_size)
+        else:
+            scheme = GetKratosObjectType("ResidualBasedPredictorCorrectorVelocityBossakSchemeTurbulent")(
+                bossak_alpha,
+                settings["move_mesh_strategy"].GetInt(),
+                self.domain_size)
+
+        linear_solver = GetKratosObjectType("LinearSolverFactory")(
+            settings["linear_solver_settings"])
+
+        builder_and_solver = CreateBlockBuilderAndSolver(
+            linear_solver,
+            self.IsPeriodic(),
+            self.GetCommunicator())
+
+        self.solver = GetKratosObjectType("ResidualBasedNewtonRaphsonStrategy")(
+            self.monolithic_model_part,
+            scheme,
+            conv_criteria,
+            builder_and_solver,
+            settings["maximum_iterations"].GetInt(),
+            settings["compute_reactions"].GetBool(),
+            settings["reform_dofs_at_each_step"].GetBool(),
+            settings["move_mesh_flag"].GetBool())
+
+        builder_and_solver.SetEchoLevel(self.echo_level - 3)
+        self.solver.SetEchoLevel(self.echo_level - 2)
+        conv_criteria.SetEchoLevel(self.echo_level - 1)
```

Comment:
This changes the expected output for a given echo level. I'd keep the current ones. If these are not suitable, we can open a discussion to reduce the current verbosity.

Reply:
done :)